libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -656,30 +627,22 @@ + (BOOL)errorIsNetworkFailure:(NSError *)error
 
 - (void)finalizeAuthCompletion
 {
-    // Apply the credentials that will ensure there is a current user and that this
+    // Apply the credentials that will ensure there is a user and that this
     // current user as the proper credentials.
-    [[SFUserAccountManager sharedInstance] applyCredentials:self.coordinator.credentials];
-    
-    // Assign the identity data to the current user
-    NSAssert([SFUserAccountManager sharedInstance].currentUser != nil, @"Current user should not be nil at this point.");
-    [[SFUserAccountManager sharedInstance] applyIdData:self.idCoordinator.idData];
-
-    // Save the accounts
-    [[SFUserAccountManager sharedInstance] saveAccounts:nil];
-
-    // Notify the session is ready
+    SFUserAccount *user = [[SFUserAccountManager sharedInstance] applyCredentials:self.coordinator.credentials];
+    [[SFUserAccountManager sharedInstance] applyIdData:self.idCoordinator.idData forUser:user];
+        // Notify the session is ready
     [self willChangeValueForKey:@"haveValidSession"];
     [self didChangeValueForKey:@"haveValidSession"];
     NSDictionary *userInfo = nil;
-    SFUserAccount *user = [SFUserAccountManager sharedInstance].currentUser;
     if (user) {
         userInfo = @{ @"account" : user };
     }
     [self initAnalyticsManager];
     [[NSNotificationCenter defaultCenter] postNotificationName:kSFAuthenticationManagerFinishedNotification
                                                         object:self
                                                       userInfo:userInfo];
-    [self execCompletionBlocks];
```

Comment:
The [definition above](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1939/files#diff-4aedc82766c844e15a84507c44206a64R226) needs to be removed as well.  It's generating warnings.